crypto/rand/rand_lib.c

```diff
@@ -581,6 +585,38 @@ size_t rand_pool_bytes_remaining(RAND_POOL *pool)
     return pool->max_len - pool->len;
 }
 
+static int rand_pool_grow(RAND_POOL *pool, size_t len)
+{
+    if (len > pool->alloc_len - pool->len) {
+        unsigned char *p;
+        const size_t limit = pool->max_len / 2;
+        size_t newlen = pool->alloc_len;
+
+        do
+            newlen = newlen < limit ? newlen * 2 : pool->max_len;
+        while (len > newlen - pool->len);
+
+        if (pool->secure) {
+            p = OPENSSL_secure_malloc(newlen);
+            if (p == NULL) {
+                RANDerr(RAND_F_RAND_POOL_GROW, ERR_R_MALLOC_FAILURE);
+                return 0;
+            }
+            memcpy(p, pool->buffer, pool->len);
+            OPENSSL_secure_clear_free(pool->buffer, pool->alloc_len);
+        } else {
+            p = OPENSSL_realloc(pool->buffer, newlen);
```

Comment:
the new memory p[len..newlen] memory is not cleared, and if it is moved,
the old buffer is released without being cleared first.

Reply:
I'm not sure either is a problem since the non-secure memory doesn't contain interesting data but I'll fix just in case.

Note: I used zalloc for instead of figuring out the right parameters for `memset(3)`:

- secure memory is zero regardless
- the new buffer is at least twice the size of the old which makes it cheaper than it seems

Reply:
Likely not a problem, Matthias wanted me to clear_free the additional data,
but he did not insist in clearing the allocated memory, just memory sanitizer did :-(